services/src/datasets/external/natur40.rs

```
@@ -0,0 +1,290 @@
+                source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+            })?;
+        let split: Vec<_> = dataset.dataset_id.split(':').collect();
+
+        let (db_name, band_index) = match *split.as_slice() {
+            [db, band_index] => {
+                if let Ok(band_index) = band_index.parse::<isize>() {
+                    (db, band_index)
+                } else {
+                    return Err(geoengine_operators::error::Error::LoadingInfo {
+                        source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                    });
+                }
+            }
+            _ => {
+                return Err(geoengine_operators::error::Error::LoadingInfo {
+                    source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                })
+            }
+        };
+
+        let db_url = RasterDb::url_from_name(&self.base_url, db_name);
+        let dataset = self.load_dataset(db_url.clone()).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+
+        Ok(Box::new(GdalMetaDataStatic {
+            time: None,
+            params: gdal_parameters_from_dataset(&dataset, band_index, Path::new(&db_url))?,
+            result_descriptor: raster_descriptor_from_dataset(&dataset, band_index, None)?,
+        }))
+    }
+}
+
+#[async_trait]
+impl
+    MetaDataProvider<MockDatasetDataSourceLoadingInfo, VectorResultDescriptor, VectorQueryRectangle>
+    for Natur40DataProvider
+{
+    async fn meta_data(
+        &self,
+        _dataset: &DatasetId,
+    ) -> Result<
+        Box<
+            dyn MetaData<
+                MockDatasetDataSourceLoadingInfo,
+                VectorResultDescriptor,
+                VectorQueryRectangle,
+            >,
+        >,
+        geoengine_operators::error::Error,
+    > {
+        todo!()
```

This panics, maybe return an error instead?

```suggestion
        Err(geoengine_operators::error::Error::NotYetImplemented)
```

---

f65088f41f89f69c260284e8f82ee3c3b851bf5f